src/pages/Swap/LimitOrderBox/api/index.ts

```diff
@@ -0,0 +1,181 @@
+  }
+
+  const signedResult = await cowSdk.signOrder(cowOrder)
+
+  if (!signedResult || !signedResult.signature) {
+    throw new Error('Failed to sign order')
+  }
+
+  return {
+    ...order,
+    feeAmount, // from CoW APIs
+    signature: signedResult.signature,
+    signingScheme: signedResult.signingScheme,
+  }
+}
+
+export interface SubmitLimitOrderParams {
+  order: SerializableSignedLimitOrder
+  signer: Signer
+  chainId: number
+}
+
+/**
+ * Submits a limit order, producing a order ID hash.
+ */
+export async function submitLimitOrder({ order, signer, chainId }: SubmitLimitOrderParams): Promise<string> {
+  // create an order via CoW API
+  const cowSdk = CoWTrade.getCowSdk(chainId, {
+    signer,
+  })
+
+  const {
+    buyAmount,
+    buyToken,
+    receiverAddress,
+    expiresAt,
+    sellAmount,
+    sellToken,
+    kind,
+    signature,
+    signingScheme,
+    userAddress,
+    feeAmount,
+    quoteId,
+  } = order
+
+  return cowSdk.cowApi.sendOrder({
+    order: {
+      quoteId,
+      buyAmount,
+      buyToken,
+      sellAmount,
+      sellToken,
+      feeAmount,
+      kind: kind === LimitOrderKind.BUY ? CoWOrderKind.BUY : CoWOrderKind.SELL,
+      receiver: receiverAddress, // the account that will receive the order
+      validTo: expiresAt,
+      partiallyFillable: false, // @todo: support partially fillable orders
+      signature,
+      signingScheme,
+    },
+    owner: userAddress,
+  })
+}
+
+/**
+ * Returns the vault relayer contract address for the given chain.
+ * ERC20 tokens must approve this address.
+ * @param chainId The chain Id
+ * @returns The vault relayer address
+ */
+export function getVaultRelayerAddress(chainId: ChainId) {
+  const GPv2VaultRelayer = contractNetworks.GPv2VaultRelayer as Record<
+    ChainId,
+    Record<'transactionHash' | 'address', string>
+  >
+
+  return GPv2VaultRelayer[chainId]?.address
+}
+
+export async function getOwnerOrders(chainId: ChainId, owner: string) {
+  const cowSdk = CoWTrade.getCowSdk(chainId)
+  return cowSdk.cowApi.getOrders({ owner })
+}
+
+export async function deleteOpenOrders(chainId: ChainId, uid: string, signer: Signer) {
+  const cowSdk = CoWTrade.getCowSdk(chainId, { signer })
+  const { signature } = await cowSdk.signOrderCancellation(uid)
+
+  // @ts-ignore POLYGON is not supported now by CoW SDK
+  const url = `${cowSdk.cowApi.API_BASE_URL[chainId]}/v1/orders/${uid}`
+  try {
+    const response = await fetch(url, {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ signature, signingScheme: 'eip712' }),
+    })
+    const data = await response.json()
+    if (!response.ok) {
+      // get error message from body or default to response status
+      const error = (data && data.message) || response.status
+      return Promise.reject(error)
+    }
+    return data
+  } catch (error) {
+    console.error('There was an error!', error)
+  }
+}
```

Comment:
yes

Reply:
But cowSwap orders are posted with near immediate expiry time. So it will be either executed or cancelled immediately when the time expires. So there is no much chance to delete/cancel.